src/main/java/bio/terra/service/dataset/flight/upgrade/enableSecureMonitoring/EnableSecureMonitoringFlight.java

```diff
@@ -0,0 +1,55 @@
+package bio.terra.service.dataset.flight.upgrade.enableSecureMonitoring;
+
+import bio.terra.common.CloudPlatformWrapper;
+import bio.terra.common.exception.FeatureNotImplementedException;
+import bio.terra.common.iam.AuthenticatedUserRequest;
+import bio.terra.service.dataset.Dataset;
+import bio.terra.service.dataset.DatasetDao;
+import bio.terra.service.dataset.DatasetService;
+import bio.terra.service.dataset.flight.LockDatasetStep;
+import bio.terra.service.dataset.flight.UnlockDatasetStep;
+import bio.terra.service.job.JobMapKeys;
+import bio.terra.service.policy.PolicyService;
+import bio.terra.service.resourcemanagement.BufferService;
+import bio.terra.service.snapshot.SnapshotService;
+import bio.terra.stairway.Flight;
+import bio.terra.stairway.FlightMap;
+import java.util.UUID;
+import org.springframework.context.ApplicationContext;
+
+public class EnableSecureMonitoringFlight extends Flight {
+
+  public EnableSecureMonitoringFlight(FlightMap inputParameters, Object applicationContext) {
+    super(inputParameters, applicationContext);
+
+    ApplicationContext appContext = (ApplicationContext) applicationContext;
+
+    DatasetService datasetService = appContext.getBean(DatasetService.class);
+    DatasetDao datasetDao = appContext.getBean(DatasetDao.class);
+    SnapshotService snapshotService = appContext.getBean(SnapshotService.class);
+    BufferService bufferService = appContext.getBean(BufferService.class);
+    PolicyService policyService = appContext.getBean(PolicyService.class);
+
+    AuthenticatedUserRequest userReq =
+        inputParameters.get(JobMapKeys.AUTH_USER_INFO.getKeyName(), AuthenticatedUserRequest.class);
+
+    UUID datasetId = inputParameters.get(JobMapKeys.DATASET_ID.getKeyName(), UUID.class);
+
+    Dataset dataset = datasetService.retrieve(datasetId);
+    CloudPlatformWrapper platform =
+        CloudPlatformWrapper.of(dataset.getDatasetSummary().getStorageCloudPlatform());
+    if (platform.isGcp()) {
+      addStep(new LockDatasetStep(datasetService, datasetId, false));
+      addStep(new EnableSecureMonitoringFlipFlagStep(datasetId, datasetDao, userReq));
+      addStep(
+          new EnableSecureMonitoringRefolderGcpProjectsStep(
+              dataset, snapshotService, bufferService, userReq));
+      addStep(
+          new EnableSecureMonitoringCreateSourceDatasetAndSnapshotsTpsPolicyStep(
+              datasetId, snapshotService, policyService, userReq));
+      addStep(new UnlockDatasetStep(datasetService, datasetId, false));
+    } else {
+      throw new FeatureNotImplementedException("Secure monitoring is only supported on GCP");
```

Comment:
This isn't strictly true.  It's supported on Azure but we don't want to add the patch  to set it to true

Reply:
True - I've added a little more nuance to this message.